### ISSUE TITLE
Bug with getter and setter methods

### ISSUE THREAD

**Issue by ravenjohn** [cwingrav/auto-bind-inheritance#1]
The code is executing the getter method but the class is not yet initialized.

```js
const autoBindI = require('auto-bind-inheritance');

class A {
   constructor() {
       autoBindI(this);
   }
}

class B extends A {
    constructor() {
         super();
         this.foo = [];
    }

    // when `autoBindI` detects this method, this.foo is still undefined
    // `self[key] = val.bind(self);` will now result in
    // "Cannot read property '0' of undefined"
    get bar() {
        return this.foo[0];
    }
}

new B();
```

I think it should skip getter and setter methods.

**Comment by cwingrav** [comment]
Thanks for the post! I appreciate it. Since I don't use getter/setters, this didn't occur to me. I had to look it up. When I call `self[key]= <my bound function>`, it evaluates `self[key]` before it is set, which is a bummer since that causes the error. If it would just set be set, without calling, then there would not be an issue. 

I'm adding in an exception handler to catch this and check for getter/setter issue. To fix, try one of these:
1) initialize the getter/setter inside the getter/setter (not ideal)
2) call autoBindI at end of final child (not ideal)
3) manually call autoBindI
4) avoid getter/setters

**Comment by cwingrav** [comment]
Fixed with #3b75d5d.
